src/new-components/button-link/button-link.test.tsx

```diff
@@ -258,4 +264,82 @@ describe('ButtonLink', () => {
             expect(buttonLink.textContent).toMatchInlineSnapshot(`"😄"`)
         })
     })
+
+    describe('when loading={true}', () => {
```

Comment:
I wonder what the use case for this is. Being a link users can right-click and open the target link in a new tab, skipping its onClick prop. If we can't think of a use case then maybe we don't expose it as an option yet?

Reply:
Thinking on this some more, we may be generating a link that takes a while to resolve. What do you think about disallowing the onClick prop instead while allowing loading?

Reply:
I thought about this probably making no sense for links, but I decided not to second guess what it would be useful for. Maybe I can take it off the table for links, and we only add it if it makes sense.

Reply:
Nah, I'll leave it. The Braid `ButtonLink` has it too. It won't hurt.